server/src/routes/api/classes.js

```diff
@@ -17,29 +17,48 @@ var photosUpload = upload.fields([
 
 router.get('/', async (req, res, next) => {
   try {
-    const { hostId } = req.query;
+    const limitValue = parseInt(req.query.limit);
+    const skipValue = parseInt(req.query.skip);
+    
+    if (skipValue < 0) return res.status(400).json({message: 'This is already the most previous page'})
+
+    const { hostId, category, city, date } = req.query;
     let classes = [];
+
     if (hostId) {
       // classes by the specified host
-      classes = await Class.find({ host: new ObjectId(hostId) }).populate('host');
+      classes = await Class.find({host: new ObjectId(hostId)}).populate('host').skip(skipValue).limit(limitValue)
+    }  
+    else if (category) {
```

Comment:
If I understand this code directly I can no filter both for category and dates? This seems broken kind of. A user wants to have multiple filter active at the same time.